httpdocs/ajax/server/session.php

```diff
@@ -16,13 +16,15 @@ function return_url($url) {
   error("can't connect to MySQL database: $mysqli->connect_error");
 $mysqli->set_charset('utf8');
 $url = $mysqli->escape_string($_GET['url']);
+$branch = basename(dirname(dirname(dirname(dirname(__FILE__)))));
+$extra_condition = "id IN (SELECT server.id FROM server JOIN server_branch ON server.id=server_branch.id WHERE branch=\"$branch\")";
 # search for a dedicated server first
-$query = "SELECT url FROM server WHERE `load` < 100 AND id IN (SELECT server FROM repository WHERE \"$url%\" LIKE CONCAT(url, '%')) ORDER BY `load` LIMIT 1";
+$query = "SELECT url FROM server WHERE `load` < 100 AND id IN (SELECT server FROM repository WHERE \"$url%\" LIKE CONCAT(url, '%')) AND $extra_condition ORDER BY `load` LIMIT 1";

```

Comment:
This is not optimal, we should use a JOIN on server in the main query instead of an extra condition that contains a SELECT.